service/api/prettyprint.go

```diff
@@ -403,14 +409,34 @@ func PrettyExamineMemory(address uintptr, memArea []byte, format byte) string {
 
 	var b strings.Builder
 	w := tabwriter.NewWriter(&b, 0, 0, 3, ' ', 0)
+
 	for i := 0; i < rows; i++ {
 		fmt.Fprintf(w, addrFmt, address)
-		for j := 0; j < cols && i*cols+j < l; j++ {
-			fmt.Fprintf(w, colFormat, memArea[i*cols+j])
+
+		for j := 0; j < cols; j++ {
+			offset := i*(cols*colBytes) + j*colBytes
+			if offset+colBytes <= len(memArea) {
+				n := byteArrayToUInt64(memArea[offset:offset+colBytes], isLittleEndian)
+				fmt.Fprintf(w, colFormat, n)
+			}
 		}
 		fmt.Fprintln(w, "")
 		address += uintptr(cols)
 	}
 	w.Flush()
 	return b.String()
 }
+
+func byteArrayToUInt64(buf []byte, isLittleEndian bool) uint64 {
```

Comment:
This can be replaced with `binary.<Endian>.Uint64`, no?

Reply:
`binary.<Endian>.Uint64` only works if `len(buf)==8`.